docs/DeveloperGuide.adoc

```diff
@@ -295,61 +295,147 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
+|`* * *` |user |set reminders for internship deadlines/appointments| make sure I do not miss any internship opportunities
 
-|`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
+|`* *` |user |be able to add companies I wish to apply to in a wish-list |apply to them when the window opens
 
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
+|`* *` |self-reflecting user |see at which stage my internship application failed |get a better idea of what to improve on
+
+|`* *` |future job seeker|use this program to easily reference successful applications |apply them to future endeavours
+
+|`* *` |disorganised user |store my cover letters |easily refer to them when applying for internships
+
+|`* *` |user |give a rating to each internship based on my preference |easily decide which internship to prioritise
+
+|`*` |frequent interviewee |maintain a checklist of questions to ask the interviewer |
+
+|`*` |first-time internship seeker |use the program as a guide to internship applications |learn how to start applying for an internship
 |=======================================================================
 
 _{More to be added}_
 
 [appendix]
 == Use Cases
 
-(For all use cases below, the *System* is the `AddressBook` and the *Actor* is the `user`, unless specified otherwise)
+(For all use cases below, the *System* is the `Internship Diary` (InternDiary) and the *Actor* is the `user`, unless specified otherwise)
 
 [discrete]
-=== Use case: Delete person
+=== Use case: UC1 - Find Internship Application
 
 *MSS*
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1.  User requests to find a specific Internship Application by company name.
+2.  InternDiary shows the Internship Application with all its corresponding fields.
 +
 Use case ends.
 
 *Extensions*
 
 [none]
-* 2a. The list is empty.
+* 2a. No Internship Application is shown.
 +
 Use case ends.
 
-* 3a. The given index is invalid.
+[discrete]
+=== Use case: UC2 - Edit Internship Application
+
+*MSS*
+
+1. User pass:q[<u>find Internship Application UC1</u>].
+2. User requests to edit the fields of the Internship Application.
+3. InternDiary updates the new fields of the Internship Application.
 +
+Use case ends
+
+*Extensions*
+
 [none]
-** 3a1. AddressBook shows an error message.
+* 2a. The given index is invalid.
+[none]
+** 2a1. InternDiary shows an error message
 +
-Use case resumes at step 2.
+Use case resumes at step 1
+
+[discrete]
+=== Use case: UC3 - Prioritise Internship Application
+
+*MSS*
+
+1. User pass:q[<u>find Internship Application UC1</u>].
+2. User requests to prioritise the Internship Application.
+3. InternDiary updates the priority level of the Internship Application.
++
+Use case ends
```

Comment:
Consider adding extension for UC3 when the application already has the specified priority level